modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/CSPComplianceCheck.java

```diff
@@ -162,24 +181,70 @@ private void _checkIllegalTags(
 
 				int lineNumber = getLineNumber(content, x);
 
+				if (fileName.endsWith(".ftl") &&
+					_illegalTagAuiReplacements.contains(tagName)) {
+
+					if (_illegalTagAuiReplacements.contains(tagName)) {
+						_checkMissingAttribute(
+							fileName, tagName, "${nonceAttribute}", tagString,
+							lineNumber);
+					}
+				}
+				else if (fileName.endsWith(".vm") &&
+						 _illegalTagAuiReplacements.contains(tagName)) {
+
+					_checkMissingAttribute(
+						fileName, tagName, "$nonceAttribute", tagString,
+						lineNumber);
+				}
+			}
```

Comment:
We don't need to enter this loop at all unless the following conditions are met: 
1. `fileName` must end with ".ftl" or ".vm"
2. `_illegalTagAuiReplacements` must contain `tagName`
Pull this loop into its own `_check*` method and wrap it in a conditional.